### PR TITLE
Update kokkos min to 4.1

### PR DESCRIPTION
**Pull request by kwitaechong**
Update Kokkos minimum requirement to 4.1 for replacing pushRegion/popRegion to ScopedRegion

### CODE REVIEW

File: docker/Dockerfile.hipcc
Comment (dalg24) [inline docker/Dockerfile.hipcc:1]: This is not sufficient.  Kokkos 4.X requires a minimum ROCm version of 5.2

Comment (streeve) [reply docker/Dockerfile.hipcc:1]: Fixed in #735

Comment (streeve) [reply docker/Dockerfile.hipcc:1]: Fixed here instead

**Review by streeve — APPROVED** [review]

**Comment by streeve** [comment]
@dalg24 are you okay to merge this?

**Comment by dalg24** [comment]
> @dalg24 are you okay to merge this?

yes

**Pull request merged**